source/storage/apis-1.5.rst

```diff
@@ -795,6 +833,64 @@ collection, this technique should always be combined with the
         next_offset = r.headers.get("X-Weave-Next-Offset")
 
 
+.. _syncstorage_batch_upload:
+
+Example: uploading a large batch of items 
+-----------------------------------------
+
+The syncstorage server allows several upload requests to be combined into a
+single "batch" so that they all become visible to other clients as a single
+atommic unit.  This is achieve by using the **batch** and **commit** parameters
+on the upload request.
+
+Clients should begin by issuing a **POST /storage/<collection>?batch=true**
+request, which will accept items for upload and issue a new batch id in the
+response body.
+
+To add more items to the batch, make additional **POST** requests to the
+collection using the value of **batch** from the response body as the
+**batch** query parameter.
+
+When the final items have been uploaded, pass the **commit** query parameter
+to the **POST** request.  This will finalize the batch and make the uploaded
+items visible to other clients.  The last-modified time of the collection,
+as well as of all items included as part of the batch, will be incremeted to
+the timestap of this final **commit** request.
+
+To guard against other clients making concurrent changes to the
+collection, this technique should always be combined with the
+**X-If-Unmodified-Since** header as shown below::
```

Comment:
This pokes at something fairly fundamental. I think this should be phrased as "To guard against other clients concurrently committing changes to the collection" — `X-I-U-S` doesn't, can't, and shouldn't detect other in-progress batches.